backends/p4tools/modules/testgen/core/small_step/table_stepper.cpp

```diff
@@ -578,6 +576,15 @@ TableStepper::TableStepper(ExprStepper *stepper, const IR::P4Table *table)
         const auto *action = table->getActionList()->actionList.at(index);
         properties.actionIdMap.emplace(action->controlPlaneName(), index);
     }
+
+    // Set the appropriate properties when the table is immutable, meaning it has constant entries.
+    TableUtils::checkTableImmutability(*table, properties);
+
+    // If the table is in the set of entities to skip, we set it immutable.
+    // P4Testgen will not add a control plane entry for this table.
+    if (TestgenOptions::get().skippedControlPlaneEntities.count(properties.tableName) != 0U) {
+        properties.tableIsImmutable = true;
```

Comment:
checkTableImmutability is setting this property based on what's passed in instead? I guess not technically part of this PR, but not a great way to go... Can you just make it 'IsTableImmutable' and then set the property in the same way you do here?

Reply:
The current implementation for `checkTableImmutability` is this one: 
```cpp
void checkTableImmutability(const IR::P4Table &table, TableProperties &properties) {
    bool isConstant = false;
    const auto *entriesAnnotation = table.properties->getProperty("entries");
    if (entriesAnnotation != nullptr) {
        isConstant = entriesAnnotation->isConstant;
    }
    // Also check if the table is invisible to the control plane.
    // This also implies that it cannot be modified.
    properties.tableIsImmutable = isConstant || table.getAnnotation("hidden") != nullptr;
    const auto *defaultAction = table.properties->getProperty("default_action");
    CHECK_NULL(defaultAction);
    properties.defaultIsImmutable = defaultAction->isConstant;
}
```
It also sets `defaultIsImmutable`. The main reason the function was constructed this way is because other extensions may pass in an extended `TableProperties` class. For example, `TofinoTableProperties`.  

The implementation can definitely be improved, I can try to refactor this in a separate PR.

Reply:
Ah, I see. Ok. I would defer to you on how clear the implementation is versus the breadth that it needs to support. Looks pretty fine to me tbh. It just threw me off that they were setting the same value and looked totally different.